examples/flow.cpp

```diff
@@ -184,15 +252,17 @@ try
         ioConfig->overrideRestartWriteInterval((size_t)output_interval);
     }
 
-    Opm::BlackoilOutputWriter outputWriter(cGrid,
-                                           param,
-                                           eclipseState,
-                                           pu );
+    const PhaseUsage pu = Opm::phaseUsageFromDeck(deck);
+    Opm::BlackoilOutputWriter outputWriter(grid, param, eclipseState, pu );
 
     // Rock and fluid init
-    props.reset(new BlackoilPropertiesFromDeck(deck, eclipseState, *grid->c_grid(), param));
-    new_props.reset(new BlackoilPropsAdFromDeck(deck, eclipseState, *grid->c_grid()));
-
+    props.reset(new BlackoilPropertiesFromDeck(deck, eclipseState,
+                                               Opm::UgGridHelpers::numCells(grid),
+                                               Opm::UgGridHelpers::globalCell(grid),
+                                               Opm::UgGridHelpers::cartDims(grid),
+                                               Opm::UgGridHelpers::beginCellCentroids(grid),
+                                               Opm::UgGridHelpers::dimensions(grid), param));
+    new_props.reset(new BlackoilPropsAdFromDeck(deck, eclipseState, grid));
```

Comment:
> otherwise you can run into trouble with stack size limitations

Is that really something we need to worry about? In any case our classes typically store big data in Eigen::Array or std::vector objects, so only the managing parts are on the stack.

I cannot really see how you would build a class with  a very large stack presence -- you would need to use a large fixed-size C-type stack array or something like that, and I have trouble seeing how that could ever be useful (small ones like in the small-string optimisation yes, but not large ones). Given that, the only way to get stack trouble is by using recursive algorithms, but that can usually be avoided.